src/Symfony/Component/Notifier/Bridge/OvhCloud/OvhCloudTransport.php

```diff
@@ -0,0 +1,138 @@
+    {
+        $this->applicationKey = $applicationKey;
+        $this->applicationSecret = $applicationSecret;
+        $this->consumerKey = $consumerKey;
+        $this->serviceName = $serviceName;
+
+        parent::__construct($client, $dispatcher);
+    }
+
+    public function setHostByEndpoint(?string $endpoint): self
+    {
+        $this->host = $this->endpoints[$endpoint] ?: self::HOST;
+
+        return $this;
+    }
+
+    public function __toString(): string
+    {
+        return sprintf('ovhcloud://%s?consumer_key=%s&service_name=%s', $this->getEndpoint(), $this->consumerKey, $this->serviceName);
+    }
+
+    public function supports(MessageInterface $message): bool
+    {
+        return $message instanceof SmsMessage;
+    }
+
+    protected function doSend(MessageInterface $message): void
+    {
+        if (!$message instanceof SmsMessage) {
+            throw new LogicException(sprintf('The "%s" transport only supports instances of "%s" (instance of "%s" given).', __CLASS__, SmsMessage::class, \get_class($message)));
+        }
+
+        $endpoint = sprintf('%s/sms/%s/jobs', $this->getEndpoint(), $this->serviceName);
+
+        $content = [
+            'charset' => 'UTF-8',
+            'class' => 'flash',
+            'coding' => '8bit',
+            'message' => $message->getSubject(),
+            'receivers' => [$message->getPhone()],
+            'noStopClause' => false,
+            'priority' => 'medium',
+            'senderForResponse' => true,
+        ];
+
+        $headers['X-Ovh-Application'] = $this->applicationKey;
+
+        if (!isset($this->timeDelta)) {
+            $this->calculateTimeDelta();
+        }
+
+        $now = time() + $this->timeDelta;
+        $headers['X-Ovh-Timestamp'] = $now;
+
+        if (isset($this->consumerKey)) {
+            $toSign = $this->applicationSecret.'+'.$this->consumerKey.'+POST+'.$endpoint.'+'.json_encode($content, JSON_UNESCAPED_SLASHES).'+'.$now;
+            $headers['X-Ovh-Consumer'] = $this->consumerKey;
+            $headers['X-Ovh-Signature'] = '$1$'.sha1($toSign);
+        }
+
+        $response = $this->client->request('POST', $endpoint, [
+            'headers' => $headers,
+            'json' => $content,
+        ]);
+
+        if (200 !== $response->getStatusCode()) {
+            $error = $response->toArray(false);
+
+            throw new TransportException(sprintf('Unable to send the SMS: %s.', $error['message']), $response);
+        }
+    }
+
+    /**
+     * Calculate time delta between local machine and API's server.
+     */
+    private function calculateTimeDelta(): int
+    {
+        if (!isset($this->timeDelta)) {
+            $endpoint = sprintf('%s/auth/time', $this->getEndpoint());
+
+            $response = $this->client->request(
+                'GET',
+                $endpoint
+            );
+
+            $serverTimestamp = (int) (string) $response->getContent();
+            $this->timeDelta = $serverTimestamp - (int) time();
```

Comment:
I would not set it here, but just return the value and store it in the calling code.